spec/presenters/sort_presenter_spec.rb

```diff
@@ -0,0 +1,188 @@
+        sort_option('Relevance', 'relevance', disabled: true)
+      ].join("\n")
+
+      presenter = described_class.new(content_item(sort_options: sort_options_with_relevance), {})
+
+      expect(presenter.for_select).to eql(expected_options)
+    end
+
+    it "returns sort options with relevance enabled when keywords is not blank" do
+      expected_options = [
+        sort_option('Most viewed', 'most-viewed'),
+        sort_option('Updated (newest)', 'updated-newest'),
+        sort_option('Relevance', 'relevance', disabled: false)
+      ].join("\n")
+
+      presenter = described_class.new(content_item(sort_options: sort_options_with_relevance), "keywords" => "something not blank")
+
+      expect(presenter.for_select).to eql(expected_options)
+    end
+
+    it "returns sort options with no option selected when order is specified but does not exist in options" do
+      expected_options = [
+        sort_option('Most viewed', 'most-viewed'),
+        sort_option('Updated (newest)', 'updated-newest')
+      ].join("\n")
+
+
+      presenter = described_class.new(content_item(sort_options: sort_options_without_relevance), "order" => "option_that_does_not_exist")
+
+      expect(presenter.for_select).to eql(expected_options)
+    end
+
+    it "returns sort options with default option selected when order is not specified and default option exists" do
+      expected_options = [
+        sort_option('Most viewed', 'most-viewed'),
+        sort_option('Updated (oldest)', 'updated-oldest', selected: true)
+      ].join("\n")
+
+      presenter = described_class.new(content_item(sort_options: sort_options_with_default), {})
+
+      expect(presenter.for_select).to eql(expected_options)
+    end
+
+    it "returns sort options with option selected when order is specified and exists in options" do
+      expected_options = [
+        sort_option('Most viewed', 'most-viewed'),
+        sort_option('Updated (newest)', 'updated-newest', selected: true)
+      ].join("\n")
+
+      presenter = described_class.new(content_item(sort_options: sort_options_without_relevance), "order" => "updated-newest")
+
+      expect(presenter.for_select).to eql(expected_options)
+    end
+  end
+
+  describe "#has_options?" do
+    it "returns false if there are no options in the content item" do
+      expect(presenter_without_sort.has_options?).to be false
+    end
+
+    it "returns true if there are sort options in the content item" do
+      expect(presenter_with_sort.has_options?).to be true
+    end
+  end
+
+  describe "#has_default_option?" do
+    it "returns true if there is a default option specified in the content item" do
+      expect(presenter_with_default.has_default_option?).to be true
+    end
+
+    it "returns false if there is not a default option specified in the content item" do
+      expect(presenter_with_sort.has_default_option?).to be false
+    end
+  end
+
+  describe "#has_default_option?" do
```

Comment:
Duplicate `describe` block (first defined from line 132).

Reply:
Thanks. This has been fixed in a later commit.